### DISCUSSION TITLE
PhD Collaboration

### DISCUSSION THREAD

**Discussion by dennis-tra** [protocol/ResNetLab#12] [General]
Hi ResNetLab-Team,

Sorry for the late follow-up. I needed to think a couple of things through and wanted to get on top of your latest research - [Beyond Swapping Bits](https://research.protocol.ai/blog/2021/beyond-swapping-bits-project-review-and-preview/).

As I already mentioned in my mails, I would love to take part in your research endeavors and lend a helping hand. I think it's a great opportunity for both sides as I'm just starting out my Ph.D. I would get to work on things I'm immensely interested in and you would get an eager collaborator pushing your research further - as you encourage collaboration in many places, you presumably think the same way.

Personally, I'm interested in compiling a draft for a cohesive multi-year project plan for my Ph.D. My ideal would be to work on topics revolving around the [`Data Exchange`](https://github.com/protocol/ResNetLab#areas) or [`Content Addressing`](https://github.com/protocol/ResNetLab#areas) research areas, and for this to happen, I'll need support and guidance from you.

To get a little more concrete, the [Beyond Swapping Bits paper](https://research.protocol.ai/blog/2021/beyond-swapping-bits-project-review-and-preview/) sparked my motivation even more and made me dive deep into the [beyond-bitswap](https://github.com/protocol/beyond-bitswap) repository. I had a look around your [Enhancement RFCs](https://github.com/protocol/beyond-bitswap#enhancement-rfcs) and found the [`RFC|BB|L1/2-05: Use of super nodes and decentralized trackers`](https://github.com/protocol/beyond-bitswap/blob/master/RFC/rfcBBL1205) and [`RFC|BB|L12-01: Bitswap/Graphsync exchange messages extension and transmission choice`](https://github.com/protocol/beyond-bitswap/blob/master/RFC/rfcBBL1201) especially interesting.

So I would love to jump on a call with you and discuss the following things:

- Do you think that the scope of one of the Beyond Bitswap RFCs (also considering related tasks like a literature review, prototype implementation, evaluation) would cover a multi-year project? I expect that's not the case, so:
- Do you see related open problems (maybe outside of the Beyond Bitswap project) suitable to complement the work done in one of the above RFCs? Just speculating here: maybe there are unaddressed research questions in your [content addressing](https://github.com/protocol/ResNetLab#areas) research area that complement a Beyond Bitswap RFC.
- Do you have other ideas or areas that could be suitable for a Ph.D. project?
- Next to that, I would like to get to know you and discuss how collaboration could look like.

I hope that's concrete enough and gives you enough information to get an impression of my intentions - really looking forwarding to hearing back from you.

Best,
Dennis

**Comment by daviddias** [comment]
> Do you think that the scope of one of the Beyond Bitswap RFCs (also considering related tasks like a literature review, prototype implementation, evaluation) would cover a multi-year project? 

Absolutely, there is an ocean of potential improvements, starting with standard networking and data flows in order to optimize data transfer and load balancing of the network, new content discovery mechanisms and then jump into specific data structures that can benefit both the discovery and content transfer.

> Do you have other ideas or areas that could be suitable for a Ph.D. project?

We have, a lot! :D The collaborative notebook continues to be a great source for inspiration -- https://github.com/ipfs/notes/issues .

> Next to that, I would like to get to know you and discuss how collaboration could look like.

You are welcome to pick some time to chat, find my availability at https://calendly.com/daviddias

I recommend preparing a doc with the outline of your PhD plan in advance.

**Reply by dennis-tra** [reply]
Awesome, thanks David!

> Absolutely, there is an ocean of potential improvements, starting with standard networking and data flows in order to optimize data transfer and load balancing of the network, new content discovery mechanisms and then jump into specific data structures that can benefit both the discovery and content transfer.

That's great to hear. I wasn't quite sure about what scope you anticipated, so thanks for clarifying.

> I recommend preparing a doc with the outline of your PhD plan in advance.

I'll come prepared :) talk soon!
